### ISSUE TITLE
value is implicitly moved even though it's used afterwards

### ISSUE THREAD

**Issue by zerbina** [nim-works/nimskull#1303]
### Example

```nim
type Object = object
  has: bool

var wasDestroyed = false

proc `=destroy`(x: var Object) =
  if x.has:
    wasDestroyed = true

proc f_sink(x: sink Object) =
  # disarm the destructor:
  x.has = false

proc test() =
  var o = Object(has: true)
  try:
    try:
      f_sink(o) # must not be sunken
      raise CatchableError.newException("")
    except IOError:
      # omitting the inner try makes the code work as expected
      discard "not reached"
  except CatchableError:
    doAssert not wasDestroyed
    discard o # use `o`, preventing it from being moved earlier

test()
doAssert wasDestroyed
```

### Actual Output

The second assertion fails, as `o` was erroneously sunken into the call argument, where the destructor was then disarmed, and `wasDestructor`  thus never set to true.

### Expected Output

The program compiles and runs without error. `o` is used beyond the procedure call, so it should not be moved.

### Additional Information

The culprit is here: https://github.com/nim-works/nimskull/blob/f401e73639c2d15f2aefbad04c96199c8308b3dc/compiler/sem/mirexec.nim#L524-L533

There's no connection between the `raise` and the outer `except` in the control-flow graph, due to a missing edge between the inner and outer `except`.

**Comment by zerbina** [comment]
There was a small error in the provided example, causing it to always fail, even when `o` is not sunken. I've correct the test case now.

**Comment by zerbina** [comment]
Fixed by #1281. Control-flow between nested try/except statements/expressions is now represented correctly within the CFG.
